Doc/whatsnew/3.9.rst

```diff
@@ -673,6 +673,19 @@ Build and C API Changes
   the garbage collector respectively. (Contributed by Pablo Galindo in
   :issue:`40241`.)
 
+* Remove the following functions from the C API. Call :c:func:`PyGC_Collect`
+  explicitly to free all free lists.
+  (Contributed by Victor Stinner in :issue:`40428`.)
```

Comment:
Also, as a more minor grammar improvement, the second sentence could be started with "Instead, ", as in "Instead, call :c:func:`PyGC_Collect` explicitly to free all free lists." . If the above isn't worthwhile though, I don't think that would be worth a PR on it's own.

Reply:
The best would be to document PyGC_Collect()... but currently, this function is not documented in Doc/c-api/ !

> "why would you need to explicitly free lists that are already free?".

Is "Call PyGC_Collect() explicitly to **clear** all free lists." better?

Reply:
Feel free to propose a PR to enhance What's New In Python 3.9. English is not my first language ;-)

Reply:
@vstinner 
> Is "Call PyGC_Collect() explicitly to clear all free lists." better?

Yep, that's definitely an improvement.

> Feel free to propose a PR to enhance What's New In Python 3.9. English is not my first language ;-)

Sounds good, will do. I might also look into documenting `PyGC_Collect()`, but I'll likely require some feedback since the C-API isn't my area of expertise. Would you mind if I requested your review on it to make sure it's correct from a technical perspective?

Also, I think you have good English skills, especially for a non-native! As a native speaker who's had to go through several college English courses though, I'm glad to help where I can with the documentation; similarly to how I'd ask you or others for help w/ the C-API. :-)

Reply:
I created PR #19793 to group other C API changes related to free lists. I replace "free" with "clear" there ;-)

Reply:
@vstinner Thanks for the update. I was about to work on it tomorrow otherwise, just finishing up my finals for the semester. :-)

Reply:
> I still plan on looking into documenting PyGC_Collect, but I need to more closely analyze the implementation and existing docstring first to make sure I get the details right. I'll open a PR and request a review when it's ready (if that's okay).

Well, there is an easy way to document PyGC_Collect: it's similar to the Python gc.collect() (called without arguments), except that PyGC_Collect() does nothing if the GC is disabled.